onnxruntime/contrib_ops/cpu/quantization/quant_gemm.cc

```diff
@@ -78,14 +78,15 @@ class QGemm : protected GemmBase, public MatMulIntegerBase {
       gemm_output_buffer = std::make_unique<Tensor>(DataTypeImpl::GetType<int32_t>(), outputshape, allocator);
       gemm_output_data = gemm_output_buffer->MutableData<int32_t>();
     } else {
+      // TODO: why it needs to bypass the type check?
```

Comment:
The assumption here is that if y_scale is null, then y_zp must also be null. In this case Y's type must be float32. This is checked by the OP schema type and shape inference function defined in quantization_defs.cc.  float32 and int32 are the same size, so we can just use this space to hold quantized gemm output, and convert the result to float32 in the post process.

you can add this as comments if needed

Reply:
Perhaps, we should comment to this effect, and answer the TODO: